app/code/Magento/Paypal/Model/Api/Nvp.php

```diff
@@ -1488,7 +1488,18 @@ protected function _exportAddresses($data)
             \Magento\Framework\DataObject\Mapper::accumulateByMap($data, $shippingAddress, $this->_shippingAddressMap);
             $this->_applyStreetAndRegionWorkarounds($shippingAddress);
             // PayPal doesn't provide detailed shipping name fields, so the name will be overwritten
-            $shippingAddress->addData(['firstname'  => $data['SHIPTONAME']]);
+            $_arraydata = explode(' ', $data['SHIPTONAME'], 2);
```

Comment:
Can SHIPTONAME contain more than two names? If so splitting and grabbing the second will result in inaccurate data

Reply:
@navarr PayPal send the data for the SHIPTONAME as one string(separated by spaces). PayPal requires a first and last name on express checkout so there must be at least 2 names in the string. So what I propose ([in my most recent commit](https://github.com/magento/magento2/pull/23799/commits/8c88936245e49567ae08bf8ca5d987d7cdeaaabe)) is in the case where there are more than one first and/or last name, make an assumption that it is a first name. In the small chance where there are 2 last names, the admin can edit orders manually rather than always editing the database to insert the name(the initial issue). This issue is discussed further on [Adam Wilson's blog](http://geeklog.adamwilson.info/article/89/Magento-only-stores-shipping-address-firstname-on-Paypal-Express-orders).